### PR TITLE
Feature 22921 mig colorize field in tree views jpr

### PR DESCRIPTION
**Pull request by dz0**
https://hive.versada.eu/projects/siramis/work_packages/22921/

### CODE REVIEW

File: web_tree_dynamic_colored_field/static/description/index.html
Comment (maciej-wichowski) [inline web_tree_dynamic_colored_field/static/description/index.html]: Usually we provide our @versada.eu email

File: web_tree_dynamic_colored_field/README.rst
Comment (maciej-wichowski) [inline web_tree_dynamic_colored_field/README.rst:103]: Why this example is removed?

**Review by maciej-wichowski — APPROVED** [review]
FYI, when pushing not migrated module to `17.0-versada`, you should set `"installable": False` in `__manifest__.py`. Branch `17.0-versada` is for our internal use, so if someone would add it into his project, Odoo would reject modules on init due to module version mismatch.

File: web_tree_dynamic_colored_field/README.rst
Comment (dz0) [reply web_tree_dynamic_colored_field/README.rst:103]: I thought it refers to the deprecated functionality... (Oct 14, 2022)
https://github.com/OCA/web/blob/15.0/web_tree_dynamic_colored_field/readme/ROADMAP.rst
https://github.com/OCA/web/commit/4b1f3fe09c1bd7d9d7220b9b2d8b8ba2c7e09a69

Now I noticed the removal was kind of "reverted" 3 days later (Oct 17, 2022)
https://github.com/OCA/web/commit/9994bc636b11bd1764935763e518c66c26d7cb4b

So: 1) it is confusing, 2) we didn't use it in Siramis, and 3) I didn't want to spend time on this...
May be you have ideas how to handle this?

PS: whole JS code history: https://github.com/OCA/web/commits/15.0/web_tree_dynamic_colored_field/static/src/js/web_tree_dynamic_colored_field.js

**Comment by dz0** [comment]
> FYI, when pushing not migrated module to `17.0-versada`, you should set `"installable": False` in `__manifest__.py`. Branch `17.0-versada` is for our internal use, so if someone would add it into his project, Odoo would reject modules on init due to module version mismatch.

Ok, I will know, 
but probably I can skip this in our case -- as noone ese uses `17.0-versada` web yet, and before we merge migration...

**Comment by dz0** [comment]
By the way, 
is it ok, that I squashed quite some (auxilary) commits in history till `15.0` ?
(as it was [recommended](https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests))

image

File: web_tree_dynamic_colored_field/static/description/index.html
Comment (oerp-odoo) [reply web_tree_dynamic_colored_field/static/description/index.html]: Don't modify `index.html`. It is generated automatically.

File: web_tree_dynamic_colored_field/README.rst
Comment (oerp-odoo) [reply web_tree_dynamic_colored_field/README.rst:103]: Not sure why you need to touch this..:)

**Comment by oerp-odoo** [comment]
> FYI, when pushing not migrated module to `17.0-versada`, you should set `"installable": False` in `__manifest__.py`. Branch `17.0-versada` is for our internal use, so if someone would add it into his project, Odoo would reject modules on init due to module version mismatch.

Why do we need to merge not migrated module? You could commit on your main branch yo reference your `web` branch you are working on, no need to merge anything to `17.0-versada` before you migrate. Once it is migrated, then you can merge it here and switch in our monorepo as well.

File: web_tree_dynamic_colored_field/readme/CREDITS.md
Comment (oerp-odoo) [inline web_tree_dynamic_colored_field/readme/CREDITS.md]:
```suggestion
- Versada
```

File: web_tree_dynamic_colored_field/static/description/index.html
Comment (oerp-odoo) [inline web_tree_dynamic_colored_field/static/description/index.html]: Don't modify index.html for OCA modules, it is generated from `rst` files
```suggestion
```

File: web_tree_dynamic_colored_field/static/description/index.html
Comment (dz0) [reply web_tree_dynamic_colored_field/static/description/index.html]: yes, it is taken from  `CONTRIBUTORS.md`, but I got Maciejs idea :)

File: web_tree_dynamic_colored_field/README.rst
Comment (dz0) [reply web_tree_dynamic_colored_field/README.rst:103]: what do you mean by "this"?

if the docs part -- because I want docs to represent reality...

**Comment by dz0** [comment]
> > FYI, when pushing not migrated module to `17.0-versada`, you should set `"installable": False` in `__manifest__.py`. Branch `17.0-versada` is for our internal use, so if someone would add it into his project, Odoo would reject modules on init due to module version mismatch.
> 
> Why do we need to merge not migrated module? You could commit on your main branch yo reference your `web` branch you are working on, no need to merge anything to `17.0-versada` before you migrate. Once it is migrated, then you can merge it here and switch in our monorepo as well.

I thought then PR will be harder to track (and I didn't change any code "directly" in that "historical" part).... 
But yews, we could select just commits with our changes for mine-changes review...

File: web_tree_dynamic_colored_field/README.rst
Comment (maciej-wichowski) [reply web_tree_dynamic_colored_field/README.rst:103]: Well usually when we migrate OCA modules, we try to cover whole functionality, not only the one we need. Your PR might be rejected (not so bad, since we have our own branch).
So let's discuss. @oerp-odoo @DarkoNikolovski should we invest more time to migrate whole functionality?

Comment (oerp-odoo) [reply web_tree_dynamic_colored_field/README.rst:103]: If we can spare some time to properly migrate a module, then we should do it. Maybe @DarkoNikolovski can comment more on that:)

Comment (DarkoNikolovski) [reply web_tree_dynamic_colored_field/README.rst:103]: @dz0 Is it possible to migrate the missing part? If yes what would be the estimated time for that?

Comment (dz0) [reply web_tree_dynamic_colored_field/README.rst:103]: seems the functionality can't work, because of reason in [ROADMAP](https://github.com/OCA/web/blob/15.0/web_tree_dynamic_colored_field/readme/ROADMAP.rst).

tried on 17, the [schema](https://github.com/odoo/odoo/blob/17.0/odoo/import_xml.rng)  does  not allow  `colors` attr for `tree`
paveikslas

Comment (DarkoNikolovski) [reply web_tree_dynamic_colored_field/README.rst:103]: If not possible then let's continue as it is.
Wondering how it is working in v.15  or is it working at all, because I can't see `colors` there as well

File: web_tree_dynamic_colored_field/static/description/index.html
Comment (oerp-odoo) [inline web_tree_dynamic_colored_field/static/description/index.html:518]: As I said, you must not edit `index.html` for OCA modules, this is generated automatically by OCA tools. You have to modify only `rst` files which index is generated from

Comment (dz0) [reply web_tree_dynamic_colored_field/static/description/index.html:518]: hmm, I didn't -- pre-commit did :shrug:

**Review by oerp-odoo — APPROVED** [review]

**Pull request merged**
